### PR TITLE
Move ``TestDocstringCheckerReturn`` to functional tests

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

### CODE REVIEW

File: tests/functional/ext/docparams/return/missing_return_doc_Sphinx.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+def my_func(self):  # [redundant-returns-doc]
+    """This is a docstring.
+
+    :returns: One
+    """
+    return None
+
+
+def my_func(self):  # [redundant-returns-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/return/missing_return_doc_Sphinx.py:51]:
```suggestion
    """warns_sphinx_redundant_rtype_doc
```

File: tests/functional/ext/docparams/return/missing_return_doc_required.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+"""Tests for missing-return-doc and missing-return-type-doc with accept-no-returns-doc = no"""
+# pylint: disable=function-redefined, invalid-name, undefined-variable, missing-function-docstring
+# pylint: disable=unused-argument
+
+
+def my_func(self):  # [missing-return-doc, missing-return-type-doc]
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/return/missing_return_doc_required.py:6]:
```suggestion
def warns_no_docstring(self):  # [missing-return-doc, missing-return-type-doc]
```

File: tests/functional/ext/docparams/return/missing_return_doc.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+"""Tests for missing-return-doc and missing-return-type-doc"""
+# pylint: disable=function-redefined, invalid-name, undefined-variable, missing-function-docstring
+# pylint: disable=unused-argument
+
+
+def my_func(self):
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/return/missing_return_doc.py:6]:
```suggestion
def ignores_no_docstring(self):
```

File: tests/functional/ext/docparams/return/missing_return_doc.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    return False
+
+
+# Ignore unknown style
+def my_func(self):
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/return/missing_return_doc.py:11]:
```suggestion
def ignores_unknown_style(self):
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thanks a lot, amazing refactor ! 

There was an issue with suggestion done from inside commits, I had to add the suggestions later after adding a first comment with the content... This review was hell 😄

**Comment by DanielNoord** [comment]
I'll keep the changing of the docstring in mind for the other ones :)

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**